paymentsheet-example/src/androidTestDebug/java/com/stripe/android/test/core/ui/UiAutomatorText.kt

```diff
@@ -29,7 +29,15 @@ open class UiAutomatorText(
         device.findObject(selector).click()
     }
 
-    fun exists() = device.findObject(selector).exists()
+    fun exists(): Boolean {
+        val originalClassName = className
+        className = if (device.findObject(selector).exists()) {
+            originalClassName
+        } else {
+            null
+        }
+        return device.findObject(selector).exists()
+    }
```

Comment:
What exactly does this solve? I also find it weird that we change the class member as a side-effect of this method

Reply:
It will try finding UI with the className and without. Then just set the mutable className to the one it found, so that further actions can use it.

Reply:
Oh, so we don’t require it to be a `TextView` anymore, but just _any_ class? Should we just get rid of `className` then, if we relax this constraint anyway?

Reply:
I think we do for certain payment methods... I believe we want to be more specific than not, since the UI could contain the same text for different elements? 🤷‍♂️